src/midiEditorCommands.h

```diff
@@ -10,9 +10,16 @@
 
 const int MIDI_EDITOR_SECTION = 32060;
 
-// This should be called when playback starts, as otherwise, pending note off
+// Stops the notes currently being previewed.
+// If sendNoteOff is true, note off messages are sent immediately.
+// This is used to silence the preview.
+// If sendNoteOff is false, note off messages are queued but not yet sent.
+// This is used when stopping a note preview that is immediately followed by a new preview.
+void previewNotesOff(bool sendNoteOff=true);
+
+// This must be called when playback starts, as otherwise, pending note off
 // messages for OSARA MIDI preview might interfere with MIDI playback.
-void cancelMidiPreviewNotesOff();
+bool cancelMidiPreviewDoneTimer();
```

Comment:
I'm still not super fond of this function name. It's very implementation specific and code outside of midiEditorCommands shouldn't need to be concerned with that. Do you have concerns with the original name of cancelMidiPreviewNotesOff? Perhaps cancelMidiPreviewPendingNotesOff would work better for you?

Reply:
I think `cancelMidiPreviewNotesOff` is ok.

Reply:
Actually, when calling cancelMidiPreviewNotesOff followed by midiPreviewNotesOff, ain't that weird? how about cancelPendingMidiPreviewNotesOff. I think I prefer that over cancelMidiPreviewPendingNotesOff, as it more resembles the intend that a pending call of midiPreviewNotesOff is canceled.